src/Microsoft.VisualStudio.Threading/AsyncReaderWriterLock.cs

```diff
@@ -725,6 +770,20 @@ private static bool HasAnyNestedLocks(Awaiter lck, HashSet<Awaiter> lockCollecti
             return false;
         }
 
+        private static void PendingWriterLockDeadLockWatchingCallback(object? state)
+        {
+            var readerWriterLock = state as AsyncReaderWriterLock;
+            if (readerWriterLock != null)
```

Comment:
tip: rewrite the first two lines as just one:

```cs
if (state is AsyncReaderWriterLock readerWriterLock)
```

Or even better, don't no-op when the cast fails. We don't expect the cast to ever fail, right? So let's use this to call more attention to our expectations, particularly when they fail:

```cs
var readerWriterLock = (AsyncReaderWriterLock)state;
Assumes.NotNull(readerWriterLock);
```

Reply:
yeah, you are right. That is better.